### PR TITLE
Update OpenTelemetry packages to 1.8.0

### LINKED ISSUE #3086: Test OTEL retries works
Test https://www.nuget.org/packages/OpenTelemetry.Exporter.OpenTelemetryProtocol/1.8.0-beta.1 retry logic works.

Can test by delaying startup of dashboard and then ensuring that telemetry from apps is eventually correctly recorded.

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspire/issues/3086

This PR:

* Updates pages to 1.8.0
* Configures trace sampling with env var instead of code - https://github.com/open-telemetry/opentelemetry-dotnet/pull/5448
* Configures .NET OTLP retries using experimental env var - https://github.com/open-telemetry/opentelemetry-dotnet/pull/5495
* Use `UseOtlpExporter` - https://github.com/open-telemetry/opentelemetry-dotnet/pull/5452

~I tested retries and they don't appear to be working. Will contact OTEL folks and figure out what is going on.~ My delay was 20 seconds and the default retry duration is 10 seconds. 10 seconds a fine default. Dashboard should startup in that time.

TODO: Update templates and playground service defaults.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3351)

### CODE REVIEW

**Comment by JamesNK** [comment]
cc @vishweshbankwar. Do you have insight to why retries aren't working?

File: src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 using Microsoft.Extensions.DependencyInjection;
```
Comment (eerhardt) [inline src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs:3]: Don't we also need to update the PackageReference versions in the project templates as well?

https://github.com/dotnet/aspire/blob/279abee67b851a0196c5197e3b0094b651376849/src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/AspireApplication.1.ServiceDefaults.csproj#L15-L16

**Review by DamianEdwards — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]
LGTM!

**Pull request merged**